### PR TITLE
remove chained ads in case of regex-filter hit

### LINKED ISSUE #4: [Feature Request] Filter for chained/animated nodes
This one is a bit trickier to handle and explain in text. Some subtitle uploaders have decided to add incredibly intrusive animated credits. They follow a similar format:  
  
```
310
01:23:53,995 --> 01:23:54,470
S

311
01:23:54,470 --> 01:23:54,945
Su

312
01:23:54,945 --> 01:23:55,420
Sub

313
01:23:55,420 --> 01:23:55,895
Subt

314
01:23:55,895 --> 01:23:56,370
Subti

315
01:23:56,370 --> 01:23:56,845
Subtit

316
01:23:56,845 --> 01:23:57,320
Subtitl

317
01:23:57,320 --> 01:23:57,795
Subtitle

318
01:23:57,795 --> 01:23:58,270
Subtitles
U

319
01:23:58,270 --> 01:23:58,745
Subtitles By
Us

320
01:23:58,745 --> 01:23:59,220
Subtitles By
Use

321
01:23:59,220 --> 01:23:59,695
Subtitles By
User

322
01:23:59,695 --> 01:24:00,170
Subtitles By
Usern

323
01:24:00,170 --> 01:24:00,645
Subtitles By
Userna

324
01:24:00,645 --> 01:24:01,120
Subtitles By
Username
```  
  
Right now subclean can handle nodes 319 to 324 but the preceding nodes remain. A special handler will need to be written that scans for these chained nodes. I'll probably end up doing it one of two ways.  

**Option A:**    
  
1. Advertising detected at node 319  
2. Check node 318 for partial match  
3. Continue checking (and removing) previous nodes until it's unable to match  

**Option B:**  
  
1. Scan the entire file for these chained nodes  
2. Remove the entire chain regardless of the content

### PR DESCRIPTION
**Pull request by Eytan414**
I'm not really familiar with subtitles ads so I was working under the assumption that only the regex based filter yields a hit of chained nodes, so my solution is applied only for those cases.

I can also expand it to rest of the cases, my initial thought is to hold a node hit index array, and apply the fix for all those indexes right after the blacklist loop.

### CODE REVIEW

**Comment by DrKain** [comment]
Thanks for this. I'll need to do some testing locally before anything is merged.  
If you wouldn't mind, could you please set up a custom arg for this functionality? Seeing as it's an experimental feature I'd prefer it not to be enabled by default.

File: src/index.ts [hunk 1]
```diff
@@ -313,15 +313,29 @@ class SubClean {
...
-                            if (regex.exec(node.data.text)) {
+                            if (regex.exec(nodeText)) {
                                 this.log(`[Match] Advertising found in node ${index + 1} (${mark})`);
-                                if (this.args.debug) this.log('[Line] ' + node.data.text);
+                                if (this.args.debug) this.log('[Line] ' + nodeText);
                                 hits++;
-                                node.data.text = '';
+                                if(index === 0) node.data.text = '';
+                                else{
+                                    const previousNodeText = (nodes[index-1].data as Cue).text;
```
Comment (DrKain) [inline src/index.ts:332]: Is `x as Cue` necessary here?

Comment (Eytan414) [reply src/index.ts:332]: I had to import and use Cue as it got confused with other objects with "data" attribute, if it compiles and works as intended on your machine then you can remove it

**Comment by Eytan414** [comment]
> 

yes, I'll push an update soon

**Comment by Eytan414** [comment]
> Thanks for this. I'll need to do some testing locally before anything is merged. If you wouldn't mind, could you please set up a custom arg for this functionality? Seeing as it's an experimental feature I'd prefer it not to be enabled by default.

Yes, I'll push an update soon

**Comment by DrKain** [comment]
Ended up changing a few things around, mostly var names and extra logging but it should be available in the 1.5.0 release published a moment ago.  
You've been credited in the release notes too, thank you for the contribution.

**Comment by Eytan414** [comment]
Yes I just saw, thank you! I didn't like the name too, and as for the other changes they're really helpful for maintainability - great library

**Pull request merged**